### PR TITLE
Errors and warnings highlighting for UNIX platforms

### PR DESCRIPTION
**Pull request by vladislav-volkov**

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

File: cmake/features.cmake [hunk 1]
```diff
@@ -68,3 +68,5 @@ Usage: -DSELECTIVE_BUILD=ON -DSELECTIVE_BUILD_STAT=/path/*.csv" OFF
                ALLOWED_VALUES ON OFF COLLECT)
 
 set(LINKCHECKER_PY "" CACHE FILEPATH "Path to linkchecker.py for documentation check")
+
+ie_option(HIGHLIGHT_ERRORS "Highlight errors and warnings during compile time" OFF)
```
Comment (ilyachur) [inline cmake/features.cmake:72]: Maybe `ENABLE_ERROR_HIGHLIGHT` is better name.

Comment (vladislav-volkov) [reply cmake/features.cmake]: Fixed

File: cmake/features.cmake [hunk 2]
```diff
@@ -68,3 +68,5 @@ Usage: -DSELECTIVE_BUILD=ON -DSELECTIVE_BUILD_STAT=/path/*.csv" OFF
                ALLOWED_VALUES ON OFF COLLECT)
 
 set(LINKCHECKER_PY "" CACHE FILEPATH "Path to linkchecker.py for documentation check")
+
+ie_option(ENABLE_ERROR_HIGHLIGHT "Highlight errors and warnings during compile time" OFF)
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:72]: why do we need an option? It can be enabled for all cases I believe.

Comment (vladislav-volkov) [reply cmake/features.cmake:72]: For errors highlighting I use ANSI escape codes. In my opinion, it looks a little ugly in the CI or IDE logs. It useful only in terminals where ANSI escape codes are supported.

**Comment by ilya-lavrenov** [comment]
@vladislav-volkov please, resolve merge-conflicts

**Pull request merged**